impeller/aiks/paint.cc

```diff
@@ -129,7 +129,7 @@ std::shared_ptr<FilterContents> Paint::MaskBlurDescriptor::CreateMaskBlur(
   if (color_source_contents->IsSolidColor() && !color_filter) {
     return FilterContents::MakeGaussianBlur(
         FilterInput::Make(color_source_contents), sigma, sigma, style,
-        Entity::TileMode::kDecal);
+        Entity::TileMode::kDecal, color_source_contents->GetGeometry());
```

Comment:
Also need to set the geometry for the `FilterContents::MakeGaussianBlur` a few lines below.

Reply:
It seems unnecessary to make the tests pass.  Could I hit that with a gradient?

Reply:
Yeah, setting any ColorSource other than solid color should trigger it.

Reply:
I'm adding it, but it doesn't render isn't correct.  I'll have to do a followup to add that and address the rendering bug.

```c++
TEST_P(AiksTest, GaussianBlurStyleInnerGradient) {
  Canvas canvas;
  canvas.Scale(GetContentScale());

  canvas.DrawPaint({.color = Color(0.1, 0.1, 0.1, 1.0)});

  Paint paint;
  paint.color = Color::Green();
  paint.mask_blur_descriptor = Paint::MaskBlurDescriptor{
      .style = FilterContents::BlurStyle::kInner,
      .sigma = Sigma(30),
  };

  std::vector<Color> colors = {Color{0.0, 0.7, 0.8, 1.0},
                               Color{0.0, 0.8, 0.7, 1.0}};
  std::vector<Scalar> stops = {0.0, 1.0};

  paint.color_source =
      ColorSource::MakeLinearGradient({0, 0}, {200, 200}, std::move(colors),
                                      std::move(stops), Entity::TileMode::kRepeat, {});

  canvas.DrawPath(PathBuilder()
                      .MoveTo({200, 200})
                      .LineTo({300, 400})
                      .LineTo({100, 400})
                      .Close()
                      .TakePath(),
                  paint);

  // Draw another thing to make sure the clip area is reset.
  Paint red;
  red.color = Color::Red();
  canvas.DrawRect(Rect::MakeXYWH(0, 0, 200, 200), red);

  ASSERT_TRUE(OpenPlaygroundHere(canvas.EndRecordingAsPicture()));
}
```

<img width="1025" alt="Screenshot 2024-02-29 at 5 42 23 PM" src="https://github.com/flutter/engine/assets/30870216/e11f0351-5e10-43f1-8485-df7c37ad1868">

Reply:
issue filed: https://github.com/flutter/flutter/issues/144449